### PR TITLE
use distroless base image

### PR DESCRIPTION
**Pull request by matthyx**

### CODE REVIEW

**Comment by matthyx** [comment]
we don't fail more tests than before... so I guess it's fine

**Comment by Bezbran** [comment]
Why not from scratch?
We can always copy busybox into the running container for debugging.
And we can create yet another image for each build with busybox inside.

**Comment by matthyx** [comment]
> Why not from scratch? We can always copy busybox into the running container for debugging. And we can create yet another image for each build with busybox inside.

distroless is scratch + passwd + ssl certificates

you cannot copy things inside a container with `kubectl cp` without a shell and tar

File: build/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-ENV CGO_ENABLED=0
-RUN go get ./...
-RUN go build -o kube-host-sensor --ldflags "-w -s -X main.BuildVersion=$BUILD_VERSION"
-
-FROM alpine
-COPY --from=0 /app/kube-host-sensor /.
-ENTRYPOINT [ "./kube-host-sensor" ]
+ENV GO111MODULE=on CGO_ENABLED=0
+WORKDIR /work
+ARG TARGETOS TARGETARCH
```
Comment (alegrey91) [inline build/Dockerfile:5]: shouldn't we provide `TARGETOS` and `TARGETARCH` in the pipeline?

File: build/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+RUN --mount=target=. \
+    --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg \
+    GOOS=$TARGETOS GOARCH=$TARGETARCH go build -o /out/kube-host-sensor --ldflags "-w -s -X main.BuildVersion=$BUILD_VERSION" .
+
+FROM gcr.io/distroless/static-debian11:latest
+
+COPY --from=builder /out/kube-host-sensor /usr/bin/kube-host-sensor
+
+ARG image_version
```
Comment (alegrey91) [inline build/Dockerfile:16]: same as above

**Review by alegrey91 — COMMENTED** [review]
Thanks @matthyx for this great improvement!
Please, correct me if I'm wrong, but we should adjust the pipeline according to the new `ARG` items provided (`TARGETOS`, `TARGETARCH`, `image_version`), right?

File: build/Dockerfile (hunk 1 above)
Comment (matthyx) [reply build/Dockerfile:5]: Nope, this is already provided by builds during the Docker build

Comment (alegrey91) [reply build/Dockerfile:5]: ok, great, what about the `image_version` instead?

**Review by alegrey91 — APPROVED** [review]
LGTM

**Comment by dwertent** [comment]
TODO: Open a PR in the helm-chart

**Comment by matthyx** [comment]
> TODO: Open a PR in the helm-chart

https://github.com/kubescape/helm-charts/pull/259

**Pull request merged**